packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -78,4 +94,103 @@ export class ApplicationExtension<
         const {allRoutes} = params
         return allRoutes
     }
+
+    /**
+     * Called on the server to serialize the extension's asynchronous routes that will be sent to the client.
+     *
+     * @returns SerializedRouteProps - The serialized extension data.
+     * @returns null if there's no asynchronous state that needs to be serialized
+     * @throws Error if the routes have not been loaded.
+     */
+    public serializeAsyncRoutes(): SerializedRouteProps[] | null {
```

Comment:
Would there be any downside to just returning an empty array if the getRoutesAsync is not defined? Or are we using that null value somewhere as an indicator that getRoutesAsync isn't defined?

A possible alternative is to throw if you try to serialize and getRoutesAsync is not defined or has not ran?